src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -704,12 +702,51 @@ public override void GenerateScript(ScriptRunner runner)
             {
                 foreach (var kvp in _functions ?? Enumerable.Empty<KeyValuePair<string, DeclaredFunction>>())
                 {
+                    if (kvp.Value.Type == DeclaredFunction.FunctionType.TimeSeries)
+                    {
+                        runner.AddTimeSeriesDeclaration(kvp.Value);
+
+                        var script = GenerateTimeSeriesInvocationCall(kvp.Key, kvp.Value.Parameters);
+
+                        runner.AddScript(script);
+
+                        continue;
+                    }
+
+
                     if (kvp.Value.Type != DeclaredFunction.FunctionType.JavaScript)
                         continue;
 
                     runner.AddScript(kvp.Value.FunctionText);
                 }
             }
+
+            private static string GenerateTimeSeriesInvocationCall(string name, List<QueryExpression> parameters)
```

Comment:
why do we need this? Why not inject the call directly as CLR func?
Can we get the name that the function was called under?

Reply:
done